risc0/zkvm/src/host/receipt.rs

```diff
@@ -310,7 +310,7 @@ impl InnerReceipt {
     ) -> Result<(), VerificationError> {
         match self {
             InnerReceipt::Composite(x) => x.verify_integrity_with_context(ctx),
-            InnerReceipt::Groth16(x) => x.verify_integrity_with_context(ctx),
+            InnerReceipt::Groth16(x) => x.verify(),
```

Comment:
So I have been using `verify` to mean "verify a successful execution of the zkVM" and `verify_integrity` to "verify an execution of the zkVM, successful or not". I started doing this because most users will reach for `.verify`, since it sounds like what they want. My belief is that most users are only interested, or think about, successful executions, and that they will be surprised if `.verify` passes for an execution that failed (ended in `Faulted` for example).

In this taxonomy, this method would be `verify_integrity`.
```suggestion
            InnerReceipt::Groth16(x) => x.verify_integrity(),
```

Reply:
How about we have a single `verify()` that takes a parameter for users to decide whether success is desired? That way it's explicit and users don't have to guess which function to call.